pages/_app.tsx

```diff
@@ -1,8 +1,24 @@
-import '../styles/globals.css'
-import type { AppProps } from 'next/app'
+import "../styles/globals.css"
+import type { AppProps } from "next/app"
+import { UserContext } from "../context/UserContext"
+import { useMemo, useState } from "react"
+import { ToastContainer } from "react-toastify"
+import { BooksContext } from "../context/BooksContext"
 
 function MyApp({ Component, pageProps }: AppProps) {
-  return <Component {...pageProps} />
+  const [user, setUser] = useState(null)
+  const [books, setBooks] = useState(null)
```

Comment:
Try creating an `IBook` or `BookType` and use it as the `type`. the sample is at above.

Reply:
fixed through reworking the BooksContext